### PR TITLE
Added the version check workflow.

### PR DESCRIPTION
**Pull request by jacobcrawford**

### CODE REVIEW

File: templates/github-workflows/check_helm_chart_version.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Helm Chart Version Check
+on:
+  pull_request:
+
```
Comment (skovbakke) [inline templates/github-workflows/check_helm_chart_version.yaml:4]:
```suggestion
on:
  pull_request:
    branches: 
      - main
```
Perhaps the workflow should only trigger when there is a pull request to the main branch

File: templates/github-workflows/check_helm_chart_version.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Get Branch
+        run: |
+          raw=$(git branch -r --contains ${{ env.COMMIT }})
+          branch=${raw##*/}
+          echo BRANCH=$branch >> $GITHUB_ENV
+
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ env.BRANCH }}
```
Comment (skovbakke) [inline templates/github-workflows/check_helm_chart_version.yaml:39]: I believe this is overly complicated. When a Github action triggers on a pull request the `GITHUB_SHA` environment variable contains the last commit made to the branch being merged, and the `GITHUB_REF` variable contains the branch.  [Docs](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request)

**Review by skovbakke — CHANGES_REQUESTED** [review]
Just a couple of changes i think we can implement.

File: templates/github-workflows/check_helm_chart_version.yaml (hunk 2 above)
Comment (jacobcrawford) [reply templates/github-workflows/check_helm_chart_version.yaml:40]: The GITHUB_REF is the destination and not the source.
See "Check github ref" in this test build:
https://github.com/distributed-technologies/test/runs/2063970945?check_suite_focus=true

Comment (skovbakke) [reply templates/github-workflows/check_helm_chart_version.yaml:40]: Maybe you can use one of these then: 
- `GITHUB_HEAD_REF` Only set for pull request events. The name of the head branch. 
- `GITHUB_BASE_REF` Only set for pull request events. The name of the base branch.
[Docs](https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables)

Comment (jacobcrawford) [reply templates/github-workflows/check_helm_chart_version.yaml:40]: github.head_ref seems to point to the "source" of the pr. Thx!
https://github.com/distributed-technologies/test/runs/2064140309?check_suite_focus=true

Comment (jacobcrawford) [reply templates/github-workflows/check_helm_chart_version.yaml]: Work like a charm:

Comment (skovbakke) [reply templates/github-workflows/check_helm_chart_version.yaml]: I don't understand the example you gave. The "Check github ref" step does `echo refs/head/main`  and the result is `refs/head/main` as expected.  And this has nothing to do with the environment variable `$GITHUB_REF` or the github context `github.ref`.
I am a bit confused.

**Review by skovbakke — APPROVED** [review]
Looks good to me

File: templates/github-workflows/check_helm_chart_version.yaml (hunk 2 above)
Comment (jacobcrawford) [reply templates/github-workflows/check_helm_chart_version.yaml]: The reference is already evaluated. The code contains the bit ${{ github.ref }} which evaluates to `refs/head/main`, which is the destination and not the source of the pull request. That is why we cannot use it to fetch the version in the helm chart on the branch being merged into main. github.head_ref on the other hand references this branch, which is what I am using now.

**Review by MartinSchmidt — APPROVED** [review]

**Pull request merged**